lib/strings.nix

```diff
@@ -324,6 +325,53 @@ rec {
   */
   escapeShellArgs = concatMapStringsSep " " escapeShellArg;
 
+  /* Test whether the given name is a valid POSIX shell variable name.
+
+     Type: string -> bool
+
+     Example:
+       isValidShellVar "foo_bar000"
+       => true
+       isValidShellVar "0-bad.jpg"
+       => false
+  */
+  isValidShellVar = name: match "[a-zA-Z_][a-zA-Z0-9_]*" name != null;
+
+  /* Make shell variables from Nix variables.
+
+     This function expects an attribute set of key-value pairs to be translated
+     into shell variables. If the values are strings (or string-coercible), then
+     the generated code is POSIX sh-compatible.
+
+     Values can also be lists of strings or attribute sets of strings, which are
+     translated into Bash-style arrays and associative arrays respectively.
+     Make sure that the shell for which you're writing supports this syntax.
```

Comment:
These paragraphs switch between aspects multiple times.

Outline suggestion

`toShellVars`: short description referencing `toShellVar`. No need to go into detail.

`toShellVar`:

First paragraph: purpose of function.

Second paragraph: describe how values are mapped to variable assignments.

Third paragraph: POSIX sh compatibility.